public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -94,6 +94,9 @@ export class AppChromeService {
   }
 
   public setReturnToPrevious = (returnToPrevious: ReturnToPreviousProps) => {
+    if (!returnToPrevious.href) {
+      returnToPrevious.href = locationService.getLocation().pathname;
```

Comment:
think we should include query params here, so something like
```suggestion
      const { pathname, search } = locationService.getLocation();
      returnToPrevious.href = pathname + search;
```